src/libstd/collections/hash/map.rs

```diff
@@ -476,7 +477,7 @@ fn search_hashed_nonempty<K, V, M, F>(table: M, hash: SafeHash, mut is_match: F)
         }
 
         // If the hash doesn't match, it can't be this one..
-        if hash == full.hash() {
+        if hash == full.hash() || !compare_hashes {
```

Comment:
Should this comparison perhaps be swapped to avoid the comparison in the case that `compare_hashes` is known as a constant `false`?